RiotNSE/NotificationService.swift

```diff
@@ -259,24 +289,24 @@ class NotificationService: UNNotificationServiceExtension {
             return
         }
         
-        self.notificationContent(forEvent: event, forAccount: userAccount) { (notificationContent, ignoreBadgeUpdate) in
-            var isUnwantedNotification = false
+        self.notificationContent(forEvent: event, forAccount: userAccount) { [weak self] (notificationContent, ignoreBadgeUpdate) in
+            guard let self = self else { return }
             
-            // Modify the notification content here...
-            if let newContent = notificationContent {
-                content.title = newContent.title
-                content.subtitle = newContent.subtitle
-                content.body = newContent.body
-                content.threadIdentifier = newContent.threadIdentifier
-                content.categoryIdentifier = newContent.categoryIdentifier
-                content.userInfo = newContent.userInfo
-                content.sound = newContent.sound
-            } else {
-                //  this is an unwanted notification, mark as to be deleted when app is foregrounded again OR a new push came
+            guard let newContent = notificationContent else {
+                // We still want them removed if the NSE filtering entitlement is not available
```

Comment:
This will have no effect on the fallback notification. I wonder if we could check for the existence of filtering entitlement and behave accordingly here.

Reply:
I'm not quite sure I understand what you mean. If haven't made a mistake I preserved the original behavior: if an unwanted notification comes in and we **don't** have the entitlement then it will be shown as empty and retracted the next time the app opens, otherwise it's not going to be displayed at all.